forms-flow-api-utils/src/formsflow_api_utils/services/external/formio.py

```diff
@@ -47,94 +48,81 @@ def generate_formio_token(self):
             }
         }
         current_app.logger.info("Generate formio token using formio login API.")
-        response = requests.post(url, headers=headers, data=json.dumps(payload))
-        if response.ok:
-            form_io_token = response.headers["x-jwt-token"]
-            return form_io_token
-        raise BusinessException(
-            "Unable to get access token from formio server",
-            HTTPStatus.SERVICE_UNAVAILABLE,
-        )
+        try:
+            response = requests.post(url, headers=headers, data=json.dumps(payload))
+            if response.ok:
+                form_io_token = response.headers["x-jwt-token"]
+                return form_io_token
+            else:
+                raise BusinessException(ExternalError.ERROR_RESPONSE_RECEIVED, detail_message=response.json())
+        except requests.ConnectionError:
+            raise BusinessException(ExternalError.FORM_SERVICE_UNAVAILABLE)
 
     def create_form(self, data, formio_token):
         """Post request to formio API to create form."""
         headers = {"Content-Type": "application/json", "x-jwt-token": formio_token}
         url = f"{self.base_url}/form"
-        response = requests.post(url, headers=headers, data=json.dumps(data))
-        if response.ok:
-            return response.json()
-        raise BusinessException(response.json(), HTTPStatus.BAD_REQUEST)
-    
+        return self._invoke_service(url, headers, data=data)
+
+    @staticmethod
+    def _invoke_service(url, headers, data=None, is_get: bool = False):
+        """Invoke formio service and handle error."""
+        try:
+            if is_get:
+                response = requests.get(url, headers=headers)
+            else:
+                response = requests.post(url, headers=headers, data=json.dumps(data))
+            if response.ok:
+                return response.json()
+            else:
+                raise BusinessException(ExternalError.ERROR_RESPONSE_RECEIVED, detail_message=response.json())
+        except requests.ConnectionError:
+            raise BusinessException(ExternalError.FORM_SERVICE_UNAVAILABLE)
+
     def update_form(self, form_id, data, formio_token):
         """Put request to formio API to update form."""
         headers = {"Content-Type": "application/json", "x-jwt-token": formio_token}
         url = f"{self.base_url}/form/{form_id}"
-        response = requests.put(url, headers=headers, data=json.dumps(data))
-        if response.ok:
-            return response.json()
-        raise BusinessException(response.json(), HTTPStatus.BAD_REQUEST)
+        return self._invoke_service(url, headers, data=data)
```

Comment:
`_invoke_service()` method also requires a PUT request for the update form.

Reply:
good catch, updated the PR